src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,71 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+enum FilterType {
+  all = 'all',
+  active = 'active',
+  completed = 'completed',
+}
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+type Props = {
+  query: string;
+  setQuery: (query: string) => void;
+  completedFilter: FilterType | null;
```

Comment:
The `completedFilter` prop is typed as `FilterType | null`, but the component defaults to `FilterType.all` when `completedFilter` is `null`. Consider initializing `completedFilter` with `FilterType.all` in the parent component to avoid handling `null` values here.